packages/@aws-cdk/aws-lambda-golang/lib/bundling.ts

```diff
@@ -0,0 +1,169 @@
+import * as os from 'os';
+import * as path from 'path';
+import { AssetCode, Code, Runtime } from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { BundlingOptions } from './types';
+import { exec, findUp, getGoBuildVersion } from './util';
+
+/**
+ * Options for bundling
+ */
+export interface BundlingProps extends BundlingOptions {
+  /**
+   * Path to go.mod file
+   */
+  readonly modFilePath: string;
```

Comment:
```suggestion
  readonly moduleDir: string;
```

Reply:
Also updated with some additional logic that verifies that you are providing a directory, and if you are providing a file, then it validates that it is a `go.mod` file.